panel/widgets/tqdm_indicator.py

```diff
@@ -0,0 +1,139 @@
+# I did not put it into the indicators.py file yet as it raises errors due to circular imports I
+# believe
+import panel as pn
+import param
+from tqdm.asyncio import tqdm as _tqdm
+
+
+EMPTY_TEXT = " "  # Hack: If set to None or "" the height of the text_pane is too little
+
+MARGIN = {
+    "text_pane": {"column": (5, 10, 0, 10), "row": (0, 10, 0, 10)},
+    "progress_indicator": {"column": (0, 10, 5, 10), "row": (12, 10, 0, 10)},
+}
+
+# I cannot have it inherit from pn.indicators.BaseIndicator as that raises an exception
+class Tqdm(pn.viewable.Viewer):
+    value = param.Integer(
+        default=0,
+        bounds=(0, None),
+        doc="""
+        The current value of the progress bar. If set to None the progress
+        bar will be indeterminate and animate depending on the active
+        parameter.""",
+    )
```

Comment:
Please be consistent about the formatting.

```suggestion
    value = param.Integer(default=0, bounds=(0, None), doc="""
        The current value of the progress bar. If set to None the progress
        bar will be indeterminate and animate depending on the active
        parameter.""")
```

Reply:
Black :-)

Reply:
No 😄

Reply:
Well maybe eventually.